node_modules/oae-core/documentpreview/js/documentpreview.js

```diff
@@ -84,86 +171,85 @@ define(['jquery', 'underscore', 'oae.core', 'lazyload'], function($, _, oae) {
             // Indicates whether or not we attempted to load this page previously
             _isRetryAttempt = _isRetryAttempt || false;
 
-            // Don't reload the page if it has already been loaded
-            if (page.$el) {
+            // Don't reload the page if all of its components have already been loaded
+            if (page.$el && page.$html && page.$text) {
                 return callback();
             }
+
+            // Variables to track progress of HTML and text content loads
+            var htmlDeferred = new $.Deferred();
```

Comment:
The problem with that approach is that it serializes the two AJAX requests, thus adding an additional round trip delay to page rendering. It seems like that might be a pretty high cost (especially for lengthy documents) as the document preview content page is pretty slow to load and render already.

Accessing `oae.gatech.edu` from Atlanta shows about 400-600ms of latency per request, so this would add about half a second of delay for each page in the document.